src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -387,6 +389,70 @@ private EmailWrapper generateFeedbackSessionEmailBaseForInstructorReminders(
                 EmailType.FEEDBACK_UNPUBLISHED.getSubject());
     }
 
+    public EmailWrapper generateFeedbackSessionResendLinksEmail(String userEmail) {
+        Date endTime = new Date();
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.MONTH, -1);
+        Date startTime = calendar.getTime();
+        String template = EmailTemplates.USER_FEEDBACK_SESSION.replace("${status}", FEEDBACK_STATUS_SESSION_OPEN);
+        String feedbackAction = FEEDBACK_ACTION_SUBMIT;
+        String subject = EmailType.FEEDBACK_LINKS_RESENT.getSubject();
+        StringBuffer buffer = new StringBuffer();
+
+        List<FeedbackSessionAttributes> sessions = fsLogic.getAllOpenFeedbackSessions(startTime, endTime);
+        for (FeedbackSessionAttributes session : sessions) {
+            // Do not send links to feedback sessions that are closed but not yet published
+            if (session.isClosed() && !session.isPublished()) {
+                continue;
+            }
+
+            CourseAttributes course = coursesLogic.getCourse(session.getCourseId());
+            StudentAttributes student = studentsLogic.getStudentForEmail(course.getId(), userEmail);
+            if (student != null) {
+                String additionalContactInformation = HTML_NO_ACTION_REQUIRED
+                        + getAdditionalContactInformationFragment(course);
+
+                String submitUrl = Config.getAppUrl(Const.ActionURIs.STUDENT_FEEDBACK_SUBMISSION_EDIT_PAGE)
+                        .withCourseId(course.getId())
+                        .withSessionName(session.getFeedbackSessionName())
+                        .withStudentEmail(userEmail)
+                        .toAbsoluteString();
+
+                String reportUrl = Config.getAppUrl(Const.ActionURIs.STUDENT_FEEDBACK_RESULTS_PAGE)
+                        .withCourseId(course.getId())
+                        .withSessionName(session.getFeedbackSessionName())
+                        .withStudentEmail(userEmail)
+                        .toAbsoluteString();
+
+                if (session.isClosed()) {
+                    submitUrl = "";
+                }
```

Comment:
This should be put together with the declaration of `submitUrl` above, avoiding calling `getAppUrl` if the action is not applicable.

Reply:
Anyway, what happens to `reportUrl` if the session is open and not published?